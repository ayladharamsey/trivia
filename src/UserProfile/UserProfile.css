.userContainer {
    background-color: #0B0C10; 
    border: 3px solid #66FCF1;
    border-radius: 5px;
    width: 95vw;
    height: 20vh;
    justify-content: space-evenly;
    display: flex;
    margin-top: 30px;
    margin-left: 40px;
}

.right {
    text-align: left;
}

h2 {
    font-family: 'Courier New', Courier, monospace;
    font-size: 1.3em;
}