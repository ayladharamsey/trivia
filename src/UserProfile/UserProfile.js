import React from 'react';
import PropTypes from 'prop-types'
import './UserProfile.css';
import { Link } from 'react-router-dom'

const UserProfile = ({ name, quote, rating }) => {
    return (
        <section>
            <img alt='profile'></img>
            <h2>Welcome <span>{name}</span></h2>
            <h3>Favorite Quote: <span> {quote} </span></h3>
            <h3>Level: <span>{rating}</span></h3>
            <Link to="/movies/characters/favorites">
                <button >Favorite Characters</button> 
            </Link>
        </section>
    )
}

export default UserProfile;

UserProfile.propTypes = {
    name: PropTypes.string,
    quote: PropTypes.string,
    rating: PropTypes.string,
    viewCharacters: PropTypes.func
  }