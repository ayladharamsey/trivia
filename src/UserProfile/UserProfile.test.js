import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from './UserProfile';

describe('UserProfile', () => {
  let wrapper;


  beforeEach(() => {
    wrapper = shallow(<UserProfile 
      name = 'Ayla'
      quote = 'She likes toast'
      rating = 'master'
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});