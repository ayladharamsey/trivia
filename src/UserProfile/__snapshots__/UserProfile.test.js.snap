// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapshot 1`] = `
<section
  className="userContainer"
>
  <div
    className="left"
  >
    <h2>
      Welcome 
      <span>
        Ayla
      </span>
    </h2>
  </div>
  <div
    className="right"
  >
    <h3>
      Favorite Quote: 
      <span>
         
        She likes toast
         
      </span>
    </h3>
    <h3>
      Level: 
      <span>
        master
      </span>
    </h3>
    <Link
      to="/movies/:id/characters/favorites"
    >
      <button>
        Favorite Characters
      </button>
    </Link>
  </div>
</section>
`;
